### DISCUSSION TITLE
Edge Functions: "Error: Network connection lost." - Vercel Ai StreamingTextResponse

### DISCUSSION THREAD

**Discussion by tycomo** [vercel/community#5933] [Help] [answered]
### Summary

In the last few days, I have started to notice an increase in the the error message "Error: Network connection lost." showing up in my application logs within Vercel for edge function execution with a status code of "xxx".   This error is shown for approximately 5% of requests.  The application itself is using OpenAI streaming where I am utilizing the vercel/ai package and returning a StreamingTextResponse as part of the edge function.

I increased tracing to try and understand scenarios where this occurs and found that it seems to be added to the logs after streaming is complete.  In this case the end user is unimpacted and the streaming response is completed but this is filling up my logs with errors making it hard to debug actual issues.  

Has anyone observed similar errors with edge functions where they are getting "Network connection lost" messages when streaming content or using vercel/ai?  I could not find much information on this specific error within any edge documentation so I am not sure how to continue debugging to resolve this issue.  My initial thought was that something was not being disposed correctly with the stream and the error is coming from the closed connection to the edge function once its complete (but this is just a guess).  There are some recent comments on a closed discussion post (https://github.com/orgs/vercel/discussions/1668) where others seem to be observing something similar but wanted to continue discussion as part of a new discussion post.

Any insight or ideas are appreciated.

**Comment by edouard-blocktool** [comment]
We're in the same boat. Started seeing these when we moved from version "ai": "^2.2.21",  to: "ai": "^2.2.32". We're checking if reverting fixes it.

**Reply by matthewdepero** [reply]
Same team, reverting to 2.2.21 did not fix the problem

**Reply by tycomo** [reply]
Glad I am not the only one!  I had tried various "ai" package versions versions to see if I could isolate any package update that caused this to appear and nothing has resolved the issue. 

What LLM API are you using in your edge functions for streaming?  I am currently using Langchain as a wrapper for OpenAI but was thinking about refactoring one of my functions to remove Langchain and just use the OpenAI API client directly to see if that resolves the errors.

**Reply by edouard-blocktool** [reply, marked as answer]
We use the Azure/OpenAI (@azure/openai). We tried the latest version of vercel/ai: "^2.2.37"

And the Network Error seem more rare. Still happen, but much less frequently.

**Reply by tycomo** [reply]
I upgraded my vercel/ai package to 2.2.37 and the frequency of the Error: Network connection lost message in my Vercel logs has gone back to _almost_ 0%.  Thanks for following up.

**Comment by amyegan** [comment]
Thanks for reaching out. The "Network connection lost" error comes from the upstream provider, and it means the edge function isn't able to connect or the connection was dropped while connecting to the endpoint. When this happens due to a Vercel issue, you would see the error [EDGE_FUNCTION_INVOCATION_FAILED](https://vercel.com/docs/errors/EDGE_FUNCTION_INVOCATION_FAILED) or [EDGE_FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/EDGE_FUNCTION_INVOCATION_TIMEOUT).

The "Network connection lost" error may appear due to clients not waiting for the entire response and abandoning the connection. If it is caused by clients going away and canceling a long-running request, then it's an expected error.

Since you're using OpenAI, it's worth checking [status.openai.com](https://status.openai.com/) for outages the next time you see that error in the logs. It could be caused by a network issue. You may also want to reach out to them for help investigating causes of [connection timeouts](https://help.openai.com/en/articles/6897186-timeout).
